lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -777,13 +779,25 @@ private void logIfAbnormallyOld(ExpiringToken token, long now, Supplier<String>
         if (log.isInfoEnabled()) {
             long age = now - token.getCreationDateMs();
             if (age > maxNormalLockAge.toMillis()) {
-                log.debug("Token refreshed which is {} ms old: {}",
-                        SafeArg.of("ageMillis", age),
-                        UnsafeArg.of("description", description.get()));
+                if (isFromAtlasTransactionWithLockedImmutable(token)) {
+                    log.warn("Token refreshed from a very long lived atlas transaction which is {} ms old: {}",
+                            SafeArg.of("ageMillis", age),
+                            UnsafeArg.of("description", description.get()));
+                } else {
+                    log.debug("Token refreshed which is {} ms old: {}",
+                            SafeArg.of("ageMillis", age),
+                            UnsafeArg.of("description", description.get()));
+                }
             }
         }
     }
 
+    private boolean isFromAtlasTransactionWithLockedImmutable(ExpiringToken token) {
+        return token.getClient() != null
+                && token.getClient().getClientId().startsWith(ATLAS_LOCK_PREFIX)
```

Comment:
Note that this _only_ works for large internal product, because of how the lock clients are injected. We might want to add support for the LockClient `atlas instance` if we want this to apply more broadly to other atlas users